### PR TITLE
Broadcasting Fixes

### PR DESCRIPTION
**Pull request by jacobdparker**

### CODE REVIEW

File: regridding/_conservative_ramshaw.py [hunk 1]
```diff
@@ -55,23 +52,39 @@ def _conservative_ramshaw(
...
-                grid_input=grid_input,
-                epsilon=epsilon,
-                weights=weights,
-            )
+    grid_static_x, grid_static_y = grid_output
+    grid_sweep_x, grid_sweep_y = grid_input
+    for axis in axes:
+        _sweep_axis(
+            # values_input=values_input,
+            # values_output=values_output,
```
Comment (byrdie) [inline regridding/_conservative_ramshaw.py:60]: Remove commented-out code

File: regridding/_conservative_ramshaw.py [hunk 2]
```diff
@@ -55,23 +52,39 @@ def _conservative_ramshaw(
...
+            epsilon=epsilon,
+            weights=weights,
+        )
+
+    grid_static_x, grid_static_y = grid_input
+    grid_sweep_x, grid_sweep_y = grid_output
+    for axis in axes:
+        _sweep_axis(
+            # values_input=values_input,
+            # values_output=values_output,
```
Comment (byrdie) [inline regridding/_conservative_ramshaw.py:77]: Remove commented-out code

File: regridding/_regrid/_regrid_from_weights.py [hunk 3]
```diff
@@ -52,8 +52,25 @@ def regrid_from_weights(
...
+    #     else:
+    #         padded_shape_input.append(1)
+    #         padded_shape_output.append(1)
+    #
+    # print(f'{padded_shape_input=}')
+    # print(f'{padded_shape_output=}')
+    #
+    # shape_input = np.broadcast_shapes(padded_shape_input, values_input.shape)
+    # print(f'{shape_input=}')
+    # values_input = np.broadcast_to(values_input, shape=shape_input, subok=True)
```
Comment (byrdie) [inline regridding/_regrid/_regrid_from_weights.py:73]: Remove commented-out code

File: regridding/_regrid/_regrid_from_weights.py [hunk 4]
```diff
@@ -66,13 +83,15 @@ def regrid_from_weights(
...
         values_output = np.zeros_like(values_input, shape=shape_output)
+
     else:
         if values_output.shape != shape_output:
             raise ValueError(f"")
         values_output.fill(0)
 
     ndim_output = len(shape_output)
     axis_output = _util._normalize_axis(axis_output, ndim=ndim_output)
+    print(f'{axis_output=}')
```
Comment (byrdie) [inline regridding/_regrid/_regrid_from_weights.py:94]: Remove print statement

File: regridding/_regrid/_regrid_from_weights.py [hunk 5]
```diff
@@ -66,13 +83,15 @@ def regrid_from_weights(
             ),
         )
         values_output = np.zeros_like(values_input, shape=shape_output)
+
```
Comment (byrdie) [inline regridding/_regrid/_regrid_from_weights.py:86]: Remove extra line

File: regridding/_regrid/_regrid_from_weights.py [hunk 6]
```diff
@@ -85,13 +104,17 @@ def regrid_from_weights(
...
     values_input = values_input.reshape(-1, *shape_input_numba)
     values_output = values_output.reshape(-1, *shape_output_numba)
 
+    weights = numba.typed.List(weights.reshape(-1))
+
     values_input = np.ascontiguousarray(values_input)
     values_output = np.ascontiguousarray(values_output)
 
+    print(f'{values_input.shape=}')
+    print(f'{values_output.shape=}')
```
Comment (byrdie) [inline regridding/_regrid/_regrid_from_weights.py:116]: Remove print statements

File: regridding/_regrid/_regrid_from_weights.py [hunk 7]
```diff
@@ -52,11 +52,17 @@ def regrid_from_weights(
     :func:`regridding.regrid_from_weights`
     """
 
-    shape_input = np.broadcast_shapes(shape_input, values_input.shape)
-    values_input = np.broadcast_to(values_input, shape=shape_input, subok=True)
```
Comment (byrdie) [inline regridding/_regrid/_regrid_from_weights.py:56]: Huh, so did this line get accidentally deleted? regardless we need to add `subok=True` to

**Review by byrdie — CHANGES_REQUESTED** [review]

File: regridding/_regrid/_regrid_from_weights.py (hunk 7 above)
Comment (jacobdparker) [reply regridding/_regrid/_regrid_from_weights.py:56]: Looks like I just moved it below (line 64) that axis normilzation which needed the broadasted shape first.

File: regridding/_regrid/_regrid_from_weights.py (hunk 5 above)
Comment (jacobdparker) [reply regridding/_regrid/_regrid_from_weights.py]: I can't tell what this is referring to.  Is it resolved?

**Review by byrdie — APPROVED** [review]

**Pull request merged**